Library/Homebrew/config.rb

```diff
@@ -36,7 +36,7 @@
 HOMEBREW_CACHE_FORMULA = HOMEBREW_CACHE/"Formula"
 
 # Where build, postinstall, and test logs of formulae are written to
-HOMEBREW_LOGS = Pathname.new(ENV["HOMEBREW_LOGS"] || "~/Library/Logs/Homebrew/").expand_path
+HOMEBREW_LOGS = Pathname.new(ENV["HOMEBREW_LOGS"])
```

Comment:
This will need to keep the `||` or it'll break on `brew update`. It doesn't need to be platform specific here, though.

Reply:
I couldn't reproduce an update error using `brew update-test --commit=HEAD^`. I'm also not sure I understand what you're asking for; are you suggesting that I leave this line unchanged instead?

Reply:
Yes leave it unchanged. It may not be reproducible by update test. When an autoupdate occurs the Bash environment variables won’t have been updated but the Ruby ones will. This may justify a separate issue (and a PR would be great) to resource all env var initalisation on a successful auto update.